### PR TITLE
Add option to change between full and simple main menu

### PR DESCRIPTION
**Pull request by PorygonZRocks**
What I did was change the check in the main menu init.lua from `PLATFORM == "Android"` to `PLATFORM == "Simple"`, and changed the init_simple.lua to change the platform to Simple.
(Ignore the rename commit, that was for an earlier idea, and I reverted it in the next one.)

### CODE REVIEW

**Comment by nerzhul** [comment]
what is the goal  here?

**Comment by PorygonZRocks** [comment]
On Android, the main menu is currently a limited UI, without the ability to have multiple singleplayer worlds or subgames, or textures. The idea with this is that it allows Android users such as myself to have access to those features, but also allows people to go back to the simple UI if they want/need by changing the "main_menu_script" option to the "init_simple.lua" file (which already existed for testing that UI on PC). It could also be set to check a separate preference, but I'm not sure how to do that.

**Comment by octacian** [comment]
:+1: I think this is definitely an improvement, however, I do think that a more user-friendly option should be added to allow switching between the simple and desktop menus. You can introduce more settings to the advanced settings menu using `builtin/settingtypes.txt` and you can then check those settings with `core.settings:get_string` (there's also `get_int`, `get_float`, and others - see documentation).

Overall though, I'd like to see the entire main menu rewritten to be tab-less, as least for Android, and as soon as I am able, I plan on doing this. Offtopic for this PR though.

**Comment by Fixer-007** [comment]
People on tablets with large screen may benefit from this.

File: builtin/settingtypes.txt [hunk 1]
```diff
@@ -1518,6 +1518,12 @@ curl_file_download_timeout (cURL file download timeout) int 300000
 #    Makes DirectX work with LuaJIT. Disable if it causes troubles.
 high_precision_fpu (High-precision FPU) bool true
 
+#    Main menu style:
+#    -   Full (default):  Multple singleplayer worlds, subgame choice, texture pack chooser, etc.
```
Comment (octacian) [inline builtin/settingtypes.txt]: Spelling mistake: "Multple" should be "Multiple"

**Comment by PorygonZRocks** [comment]
I've changed it to now use a separate setting (main_menu_style), and I also got rid of the init_simple.lua, as it is no longer needed for testing the simple menu. Thanks @octacian for the idea and explanation on how to use it.

**Comment by PorygonZRocks** [comment]
I've also added the option to the settings tab, however, the dropdown is currently out of the tab box. How do I make that box bigger?

File: builtin/mainmenu/init.lua [hunk 2]
```diff
@@ -24,15 +24,16 @@ mt_color_dark_green = "#25C191"
...
-dofile(basepath .. "fstk" .. DIR_DELIM .. "buttonbar.lua")
-dofile(basepath .. "fstk" .. DIR_DELIM .. "dialog.lua")
-dofile(basepath .. "fstk" .. DIR_DELIM .. "tabview.lua")
-dofile(basepath .. "fstk" .. DIR_DELIM .. "ui.lua")
+dofile(basepath .. DIR_DELIM .. "common" .. DIR_DELIM .. "async_event.lua")
+dofile(basepath .. DIR_DELIM .. "common" .. DIR_DELIM .. "filterlist.lua")
+dofile(basepath .. DIR_DELIM .. "fstk" .. DIR_DELIM .. "buttonbar.lua")
+dofile(basepath .. DIR_DELIM .. "fstk" .. DIR_DELIM .. "dialog.lua")
+dofile(basepath .. DIR_DELIM .. "fstk" .. DIR_DELIM .. "tabview.lua")
+dofile(basepath .. DIR_DELIM .. "fstk" .. DIR_DELIM .. "ui.lua")
```
Comment (red-001) [inline builtin/mainmenu/init.lua]: why are the last few lines changed?

File: builtin/mainmenu/tab_settings.lua [hunk 3]
```diff
@@ -247,20 +264,17 @@ local function formspec(tabview, name, tabdata)
...
+			"tableoptions[background=#00000000;highlight=#00000000;border=false]" ..
+			"table[8.33,0.7;3.5,4;shaders;" ..
+				"#888888," .. fgettext("Bump Mapping") .. "," ..
+				"#888888," .. fgettext("Tone Mapping") .. "," ..
+				"#888888," .. fgettext("Normal Mapping") .. "," ..
+				"#888888," .. fgettext("Parallax Occlusion") .. "," ..
+				"#888888," .. fgettext("Waving Water") .. "," ..
+				"#888888," .. fgettext("Waving Leaves") .. "," ..
+				"#888888," .. fgettext("Waving Plants") .. "," ..
+				";1]"
```
Comment (red-001) [inline builtin/mainmenu/tab_settings.lua]: this doesn't seem to be related to the PR

**Comment by octacian** [comment]
> I've also added the option to the settings tab, however, the dropdown is currently out of the tab box. How do I make that box bigger?

Please keep the option only in advanced settings, it's not really important enough to be top-level IMO.

File: builtin/mainmenu/init.lua (hunk 2 above)
Comment (PorygonZRocks) [inline builtin/mainmenu/init.lua]: I copy-pasted from my text editor, so it didn't properly record changes. The lines should be the same.

File: builtin/mainmenu/tab_settings.lua (hunk 3 above)
Comment (PorygonZRocks) [inline builtin/mainmenu/tab_settings.lua]: I didn't change anything there, just that Github doesn't work well with copy+paste changes.

**Comment by PorygonZRocks** [comment]
I seem to have broken something with one of my commits, so my next few are going to be a bunch of copy+pastes so I can get it working again.

**Comment by PorygonZRocks** [comment]
I have now changed it so that for Android, the default is "simple", and it is "full" for others. I have done this by adding a "default" option, which upon encoutering the lua files check the platform to determine which menu to use.

File: builtin/mainmenu/init.lua (hunk 2 above)
Comment (SmallJoker) [inline builtin/mainmenu/init.lua]: It recorded the changes with the reference of the current HEAD contents, so this partially reverts a commit between your original changes and the latest commit. Please correct this one by using the newer lines.

File: builtin/mainmenu/init.lua [hunk 4]
```diff
@@ -24,15 +24,26 @@ mt_color_dark_green = "#25C191"
...
+local mainmenustyle = core.settings:get("main_menu_style")
+local menustyle
+if mainmenustyle == "default" then
+    if PLATFORM == "Android" then
+        menustyle = "simple"
+    else
+        menustyle = "full"
+    end
+else
+    menustyle = mainmenustyle
```
Comment (SmallJoker) [inline builtin/mainmenu/init.lua]: Tabs for indents please.

File: builtin/settingtypes.txt [hunk 5]
```diff
@@ -484,10 +477,10 @@ pause_fps_max (FPS in pause menu) int 20
 viewing_range (Viewing range) int 100 20 4000
 
 #    Width component of the initial window size.
-screen_w (Screen width) int 800
+screenW (Screen width) int 800
 
 #    Height component of the initial window size.
-screen_h (Screen height) int 600
+screenH (Screen height) int 600
```
Comment (SmallJoker) [inline builtin/settingtypes.txt]: Same problem here. The PR is good when there are no copy&paste "revert" leftovers from outdated files. There's the `git rebase` function if you want to update your branch while keeping your commits.

**Comment by paramat** [comment]
Lots of unwanted reversions in settingtypes.txt.

File: builtin/settingtypes.txt (hunk 5 above)
Comment (PorygonZRocks) [inline builtin/settingtypes.txt]: I don't have command line git access.

File: builtin/mainmenu/init.lua (hunk 2 above)
Comment (PorygonZRocks) [inline builtin/mainmenu/init.lua]: It still says changed, but I merged the latest commits from base.

**Comment by paramat** [comment]
Oops this seems wrong.

**Comment by PorygonZRocks** [comment]
Merged latest. It is now up-to-date.

**Comment by paramat** [comment]
Better, but still lots of unwanted reversions in settingtypes.txt.

**Comment by PorygonZRocks** [comment]
I've fixed all the accidental reversions. 
Sorry about them, I had based my changes off files extracted from the latest release APK, and then copy+pasted.

File: builtin/mainmenu/init.lua (hunk 4 above)
Comment (PorygonZRocks) [inline builtin/mainmenu/init.lua]: Fixed.

**Comment by octacian** [comment]
:+1: Looks good, but the `.editorconfig` has to go, I dunno what editor you use, but, I for one use Atom which doesn't take such a file but instead nicely stores configuration separate from the project. Two things for the future. Don't base PRs off files extracted from latest release APK, instead just `git clone` the repository off GitHub. With a `git clone`, you can fix conflicts using `git rebase`. 

And in case you wonder how `git rebase` works, [here](https://gist.github.com/octacian/99d3996dc8835167b5d9460658a36f3e) is a short gist I made in order to remind myself how to use it properly.

**Comment by PorygonZRocks** [comment]
> I dunno what editor you use, but, I for one use Atom which doesn't take such a file

The .editorconfig file is for the Github editor, for when someone edits the files online, which is often how I commit. But now that I think about it, I generally copy and paste, so the file doesn't have much use anyway. I've removed it now.

> And in case you wonder how git rebase works, here is a short gist I made in order to remind myself how to use it properly.


Thanks for the gist.

**Comment by octacian** [comment]
Ah, I see. I'd really suggest that you look into using a "native" editor such as [Atom](https://atom.io) (written by the GitHub developers) rather than the online editor. With a bit of experience with git on the command line and Atom as an editor, you'll find PRs like this far easier to write.

With that aside, :+1:

File: builtin/mainmenu/tab_settings.lua [hunk 6]
```diff
@@ -17,6 +17,17 @@
...
+local mainmenustyle = core.settings:get("main_menu_style")
+local menustyle
+if mainmenustyle == "default" then
+    if PLATFORM == "Android" then
+        menustyle = "simple"
+    else
+        menustyle = "full"
+    end
+else
+    menustyle = mainmenustyle
```
Comment (SmallJoker) [inline builtin/mainmenu/tab_settings.lua]: Please use tabs for indents.

File: minetest.conf.example [hunk 7]
```diff
@@ -1947,6 +1947,11 @@
 #    type: bool
 # high_precision_fpu = true
 
+#    -   Full (default):  Multiple singleplayer worlds, subgame choice, texture pack chooser, etc.
+#    -   Simple: One singleplayer world, no subgame or texture pack choosers. May be necessary for smaller screens.
+#    type: enum
+main_menu_style = full
```
Comment (SmallJoker) [inline minetest.conf.example]: According to `settingtypes.txt` the default is `default`, where you use the default value `full` here. Please check what's actually the default value and correct the docs to match both the same value.

Comment (PorygonZRocks) [inline minetest.conf.example]: Sorry, I forgot to update that file. I'll fix that.

File: builtin/mainmenu/tab_settings.lua [hunk 8]
```diff
@@ -17,6 +17,17 @@
...
+local menustyle
+if mainmenustyle == "default" then
+	if PLATFORM == "Android" then
+		menustyle = "simple"
+	else
+		menustyle = "full"
+	end
+else
+	menustyle = mainmenustyle
+end
```
Comment (red-001) [inline builtin/mainmenu/tab_settings.lua]: maybe makes `menustyle` a global or put it in the minetest/core table?

File: builtin/settingtypes.txt [hunk 9]
```diff
@@ -1608,4 +1614,4 @@ modstore_listmods_url (Modstore mods list URL) string https://forum.minetest.net
 modstore_details_url (Modstore details URL) string https://forum.minetest.net/mmdb/mod/*/
 
 #    Print the engine's profiling data in regular intervals (in seconds). 0 = disable. Useful for developers.
-profiler_print_interval (Engine profiling data print interval) int 0
+profiler_print_interval (Engine profiling data print interval) int 0
```
Comment (red-001) [inline builtin/settingtypes.txt]: unrelated change

Comment (PorygonZRocks) [inline builtin/settingtypes.txt]: I don't actually see a change. I'm not sure why it says there is one.

**Comment by paramat** [comment]
:-1: Full menu will usually be unusable on Android, and Android menu will be too simple and pointless on desktop.

**Comment by rubenwardy** [comment]
> :-1: Full menu will usually be unusable on Android, and Android menu will be too simple and pointless on desktop.

Tablets shouldn't find it too hard to use the full menu. But I'm not that strongly opinioned about this PR

**Comment by PorygonZRocks** [comment]
Later today I'll deal with getting this up to date(that's what all the unrelated commits are, is it being behind). Is there a screen size variable I could use to determine the default layout, so tablets can get full by default?

**Comment by PorygonZRocks** [comment]
Rebased!

**Comment by paramat** [comment]
Unrelated commits creeping in shows your development method is wrong. You should use a separate branch for your PRs, not master, please read and follow the 'Boot camp' help section on Github about how to make a PR, it's clear and easy to follow.

**Comment by PorygonZRocks** [comment]
New pull request at [#6737](https://github.com/minetest/minetest/pull/6737). That one is based off a new branch, and has no unrelated commits. I have reverted my master branch to be even with minetest/master.

**Comment by paramat** [comment]
Thanks i appreciate it, i wasn't expecting you to change your method so fast =)

**Comment by jastevenson303** [comment]
For what it's worth, Windows 10 has a tablet mode, apparently, and when in said mode it would be useful to toggle Minetest to the Android menu (and controls).

**Comment by PorygonZRocks** [comment]
I have a new pull request at [#6737](https://github.com/minetest/minetest/pull/6737). This one was closed due to some issues (read more above). Please comment there instead, thanks :)
